nistats/utils.py

```diff
@@ -204,3 +206,159 @@ def pos_recipr(X):
     return np.where(X <= 0, 0, 1. / X)
 
 _basestring = str if py3 else basestring
+
+
+# UTILITIES FOR THE BIDS STANDARD
+def get_bids_files(main_path, file_tag='*', file_type='*', sub_label='*',
+                   file_folder='*', filters=[], parse=False, sub_folder=True,
+                   allow_other_fields=True):
+    """Search for files in a BIDS dataset following given constraints.
+
+    This utility function allows to filter files in the BIDS dataset by
+    any of the fields contained in the file names. Moreover it allows to search
+    for specific types of files or particular tags. The filter of files can
+    be inclusive or exclusive depending on the allow_other_fields.
+
+    One can get a simple list of file paths found or alternatively the parsed
+    version of the file name as a dictionary for each file found to do more
+    detailed operations on the file list.
+
+    Notice that to search in the derivatives folder, it has to be given as
+    part of the main_path. This is useful since the current convention gives
+    exactly the same inner structure to derivatives than to the main BIDS
+    dataset folder, so we can search it in the same way.
+
+    Parameters
+    ----------
+    main_path: str
+        Directory of the BIDS dataset
+
+    file_tag: str accepted by glob, optional (default: '*')
+        The final tag of the desired files. For example 'bold' if one is
+        interested in the files related to the neuroimages.
+
+    file_type: str accepted by glob, optional (default: '*')
+        The type of the desired files. For example to be able to request only
+        'nii' or 'json' files for the 'bold' tag.
+
+    sub_label: str accepted by glob, optional (default: '*')
+        Such a common filter is given as a direct option since it applies also
+        at the level of directories. the label is what follows the 'sub' field
+        in the BIDS convention as 'sub-label'.
+
+    file_folder: str accepted by glob, optional (default: '*')
```

Comment:
it could also be a user provided folder with arbitrary meaning, it was difficult to name so I settled for a general name (name of the folder that contains the file), but I am also not completely convinced by it.